hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/nfttokeninfo/NftTokenInfoCall.java

```diff
@@ -61,40 +65,54 @@ public NftTokenInfoCall(
     @Override
     protected @NonNull FullResult resultOfViewingToken(@NonNull final Token token) {
         requireNonNull(token);
-        final var nft = enhancement
-                .nativeOperations()
-                .getNft(token.tokenIdOrElse(ZERO_TOKEN_ID).tokenNum(), serialNumber);
-        final var status = nft != null ? SUCCESS : ResponseCodeEnum.INVALID_TOKEN_NFT_SERIAL_NUMBER;
-        return fullResultsFor(status, gasCalculator.viewGasRequirement(), token, nft);
+        return fullResultsFor(SUCCESS, gasCalculator.viewGasRequirement(), token);
     }
 
     @Override
     protected @NonNull FullResult viewCallResultWith(
             @NonNull final ResponseCodeEnum status, final long gasRequirement) {
-        return fullResultsFor(status, gasRequirement, Token.DEFAULT, null);
+        return fullResultsFor(status, gasRequirement, Token.DEFAULT);
     }
 
     private @NonNull FullResult fullResultsFor(
-            @NonNull final ResponseCodeEnum status,
-            final long gasRequirement,
-            @NonNull final Token token,
-            @Nullable final Nft nft) {
+            @NonNull final ResponseCodeEnum status, final long gasRequirement, @NonNull final Token token) {
         requireNonNull(status);
         requireNonNull(token);
 
         final var ledgerConfig = configuration.getConfigData(LedgerConfig.class);
         final var ledgerId = Bytes.wrap(ledgerConfig.id().toByteArray()).toString();
+        final var nft = enhancement
+                .nativeOperations()
+                .getNft(token.tokenIdOrElse(ZERO_TOKEN_ID).tokenNum(), serialNumber);
         // @Future remove to revert #9074 after modularization is completed
-        if (isStaticCall && (status != SUCCESS || nft == null)) {
+        if ((isStaticCall && (status != SUCCESS)) || nft == null) {
```

Comment:
I think this would result in CONTRACT_REVERT for non-static call in the case of not valid serial number of NFT, thus it would not match the mono response from [call-results-snapshot.json](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/resource/call-results-snapshot.json)